### ISSUE TITLE
cannot use an integer argument

### ISSUE THREAD

**Issue by 3nids** [octokit/graphql-action#7]
Using this query

```
- uses: octokit/graphql-action@v2.x
        id: doc_info
        with:
          query: |
            query doc_info($owner: String!, $repo: String!, $pr_number: Int!) {
              repository(owner: $owner, name: $repo) {
                pullRequest(number: $pr_number) {
                 commits(first: 10) {
                   edges {
                     node {
                       commit {
                         oid
                         message
                       }
                     }
                   }
                 }
               }
              }
            }
          owner: ${{ github.event.repository.owner.name }}
          repo: ${{ github.event.repository.name }}
          pr_number: $ {{ github.event.pull_request.number }}
```

I am getting this error:

```
##[error]Variable $pr_number of type Int! was provided invalid value
##[error]Node run failed with exit code 1
```

How can I deal with it?
I don't see any casting to number method in the github [context expression documentation](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions)

**Comment by gr2m** [comment]
Hmm I think the casting would need to happen within this action. Maybe after line 36

https://github.com/octokit/graphql-action/blob/fb625618dd4a08acebfead5fa39af000122e9db2/index.js#L31-L39

We could compare do something like this?

```js
result[inputName] = result[inputName] == parseInt(result[inputName], 10) ? parseInt(result[inputName], 10) : result[inputName];
```

Could you start a pull request with a failing test?

**Comment by 3nids** [comment]
test added in #8 
fix in #9 

tested and works as expected on my fork:
